packages/backend/src/modules/ArbitrumDiscoveryModule.ts

```diff
@@ -0,0 +1,66 @@
+import { Logger } from '@l2beat/backend-tools'
+import { ChainId } from '@lz/libs'
+import { providers } from 'ethers'
+
+import { Config } from '../config'
+import { BlockNumberIndexer } from '../indexers/BlockNumberIndexer'
+import { ClockIndexer } from '../indexers/ClockIndexer'
+import { DiscoveryIndexer } from '../indexers/DiscoveryIndexer'
+import { BlockchainClient } from '../peripherals/clients/BlockchainClient'
+import { BlockNumberRepository } from '../peripherals/database/BlockNumberRepository'
+import { DiscoveryRepository } from '../peripherals/database/DiscoveryRepository'
+import { IndexerStateRepository } from '../peripherals/database/IndexerStateRepository'
+import { Database } from '../peripherals/database/shared/Database'
+import { ApplicationModule } from './ApplicationModule'
+import { createDiscoveryEngine } from './EthereumDiscoveryModule'
+
+export function createArbitrumDiscoveryModule(
+  database: Database,
+  logger: Logger,
+  config: Config,
+): ApplicationModule {
+  const blockRepository = new BlockNumberRepository(database, logger)
+  const indexerRepository = new IndexerStateRepository(database, logger)
+  const discoverRepository = new DiscoveryRepository(database, logger)
+
+  const provider = new providers.JsonRpcProvider(
+    config.discovery.arbitrum.rpcUrl,
+  )
+
+  const blockchainClient = new BlockchainClient(provider, logger)
+  const discoveryEngine = createDiscoveryEngine(provider, config, 'arbitrum')
+
+  const clockIndexer = new ClockIndexer(
+    logger,
+    ChainId.ARBITRUM,
+    config.discovery.arbitrum.clockIntervalMs,
+  )
```

Comment:
ClockIndexer could be shared, but it is not necessary. Let's keep it like this for now for simplicity.

Reply:
I'll make it right as a part of module factory